### ISSUE TITLE
作者大大您好！在看完您的 resnet 训练代码后有几个地方不太明白，可能我的问题很小白，还请别介意，感谢指教

### ISSUE THREAD

**Issue by darkchii** [WZMIAOMIAO/deep-learning-for-image-processing#17]
**在您的tensorflow resnet 的 train.py 中这一段代码的作用是什么呢？（另外，很抱歉，目前我没看resnet相关论文，所以理论方面还不懂（捂脸））**
* train.py line 70
```
model = tf.keras.Sequential([feature,
                             tf.keras.layers.GlobalAvgPool2D(),
                             tf.keras.layers.Dropout(rate=0.5),
                             tf.keras.layers.Dense(1024),
                             tf.keras.layers.Dropout(rate=0.5),
                             tf.keras.layers.Dense(5),
                             tf.keras.layers.Softmax()])
```
这部分我不太明白为什么用 Sequential 再度堆叠了一个网络结构，feature 已经是 resnet 网络模型了 所以输出形状是dim=2，  umm 接下来的我不确定我的理解对不对，我的理解是下一层的全局池化层输入形状是 dim=4  ，与这里 feature 输出形状不匹配，所以无法和 feature 连接，所以我搞不太明白这段的含义，或者要如何调整呢？我的任务是使用 resnet 对一组 40*18 尺寸的灰度图像做一个二分类。盼回复！提前感谢您！

**Comment by darkchii** [comment]
然后我自己在使用中就把这段代码改为了：
```model = feature```
但跑出来结果异常值，明显不对。。。

**Comment by WZMIAOMIAO** [comment]
1）我实例化模型时使用的是feature = resnet50(num_classes=5, include_top=False)，这里的include_top=False说明实例化模型时去掉了后面的全连接层，建议好好看下model文件的代码。
2）使用灰度图像时需要改个参数，在flow_from_directory函数中有color_model参数，默认是RGB所以无论你读入的什么类型图片都给你转成RGB三通道，如果你想使用灰度图片将color_mode='grayscale'就可以了。
3）你的图像尺寸太小了，正向传播过程中长宽会缩小到原来的32倍，所以最小输入图像尺寸为32x32

**Comment by darkchii** [comment]
好的 我再试一下 感谢您的回复！
